modules/store/src/store.ts

```diff
@@ -175,20 +177,25 @@ export function select<
  * This overload is used to support spread operator with
  * fixed length tuples type in typescript 2.7
  */
-export function select<T, K = any>(
+export function select<State, Props = any, Result = any>(
+  props: Props,
   ...paths: string[]
-): (source$: Observable<T>) => Observable<K>;
-export function select<T, K>(
-  pathOrMapFn: ((state: T) => any) | string,
+): (source$: Observable<State>) => Observable<Result>;
+export function select<State, Props, K>(
+  pathOrMapFn: (state: State, props?: Props) => any | string,
+  props: Props,
   ...paths: string[]
 ) {
-  return function selectOperator(source$: Observable<T>): Observable<K> {
+  return function selectOperator(source$: Observable<State>): Observable<K> {
     let mapped$: Observable<any>;
 
     if (typeof pathOrMapFn === 'string') {
-      mapped$ = source$.pipe(pluck(pathOrMapFn, ...paths));
+      mapped$ =
+        typeof props === 'string'
```

Comment:
I think we can't support having props on a string selector e.g. `this.store.select('counter1')`, right?

Reply:
Right